rustls/tests/api.rs

```diff
@@ -2349,6 +2349,62 @@ fn server_complete_io_for_write() {
     }
 }
 
+#[derive(Default)]
+struct EofWriter<const N: usize> {
+    written: usize,
+}
+
+impl<const N: usize> std::io::Write for EofWriter<N> {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let prev = self.written;
+        self.written = N.min(self.written + buf.len());
+        Ok(self.written - prev)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        Ok(())
+    }
+}
+
+impl<const N: usize> std::io::Read for EofWriter<N> {
+    fn read(&mut self, _: &mut [u8]) -> io::Result<usize> {
+        panic!() // This is a writer, it should not be read from.
+    }
+}
+
+#[test]
+fn server_complete_io_for_write_eof() {
+    for kt in ALL_KEY_TYPES {
+        let (mut client, mut server) = make_pair(*kt);
+
+        do_handshake(&mut client, &mut server);
+
+        // Queue 20 bytes to write.
+        server
+            .writer()
+            .write_all(b"01234567890123456789")
+            .unwrap();
+        {
+            const BYTES_BEFORE_EOF: usize = 5;
+            let mut eof_writer: EofWriter<BYTES_BEFORE_EOF> = Default::default();
```

Comment:
Suggest spelling this as `let mut eof_writer = EofWriter::<BYTES_BEFORE_EOF>::default();`.

Reply:
All done.